### PR TITLE
gem structure

### PR DESCRIPTION
**Pull request by harshadixit12**
rubyvision.rb has static methods which will call respective methods from imgp.rb, bmp.rb when required. All operations done on image object, defined in image.rb.

### CODE REVIEW

File: rubyvision.gemspec [hunk 1]
```diff
@@ -0,0 +1,13 @@
+Gem::Specification.new do |s|
+  s.name        = 'rubyvision'
+  s.version     = '0.0.0'
+  s.date        = '2020-02-08'
+  s.summary     = "Image processing library in ruby"
+  s.description = "Read ans transform images"
```
Comment (pgtgrly) [inline rubyvision.gemspec]: typo

File: lib/rubyvision.rb [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+require 'bmp'
+require 'imgp'
+class Rbvision
+  def self.read(path)
+    
+  end
+
+  def self.write(path, format)
+
+  end
```
Comment (pgtgrly) [inline lib/rubyvision.rb]: Should be in Image Class

File: lib/rubyvision.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# Language: Ruby, Level: Level 3
+#wrapper class over image codecs as well as image processing module
+#contains class methods (static)
+require 'bmp'
+require 'imgp'
+class Rbvision
```
Comment (pgtgrly) [inline lib/rubyvision.rb]: Rubyvision should be the module, these functions would better fit in other classes

File: lib/rubyvision.rb [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  def self.rotate(img)
+  end
+
+  def self.kernel_transform(img, kerneltype)
+  end
+
+  def self.alpha_removal(img)
+  end
```
Comment (pgtgrly) [inline lib/rubyvision.rb]: Should belong to imageProc class

File: lib/bmp.rb [hunk 5]
```diff
@@ -0,0 +1,59 @@
+
+require 'set'
+
+class Bmp
+  def initialize(path)
+    
+  end
+
+  def readbmp(path) #returns an image object
```
Comment (pgtgrly) [inline lib/bmp.rb]: This should return an array

File: lib/bmp.rb [hunk 6]
```diff
@@ -0,0 +1,59 @@
+
+require 'set'
+
+class Bmp
```
Comment (pgtgrly) [inline lib/bmp.rb]: THis class should be called from Image Class and just be used to read and write the image

File: lib/image.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
+# Language: Ruby, Level: Level 3
+# frozen_string_literal = true
+# contains the definition of image class
+
```
Comment (pgtgrly) [inline lib/image.rb]: Require 'Bmp'

File: lib/image.rb [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+  end
+
+  def read(path)
+    
+  end
+
+  def write(path, name, format)
+  end
+  attr_reader :data
+  attr_writer :data
```
Comment (pgtgrly) [inline lib/image.rb]: use attr_accessor, it has both reader and writer

Comment (pgtgrly) [reply lib/image.rb]: attr_read :data, :height, :width, :channels
and move this statement before initialize.
We do not want other's to be able to change these values as they are property of the image and will be set by the read() function

Comment (harshadixit12) [reply lib/image.rb]: But say someone wanted to create an image, and they had pixel array, then these would need write access no?

Comment (pgtgrly) [reply lib/image.rb]: In that case, we will have another function (or the same function with conditions).

File: lib/image.rb [hunk 9]
```diff
@@ -3,7 +3,10 @@
 # contains the definition of image class
 require 'bmp'
 class Image
-
+  attr_reader :data
+  attr_reader :height
```
Comment (pgtgrly) [inline lib/image.rb]: Should be written as:
attr_reader :data, :height, :width, :channels

**Pull request merged**